File: minigit-3/py/commands/ls_files.py
"""ls-files command - List indexed files."""

import sys
from pathlib import Path


def run(args: list[str]) -> int:
    """List indexed files."""
    from utils import find_repo_root
    from index import read_index

    repo_root = find_repo_root()
    if repo_root is None:
        print("error: not a minigit repository", file=sys.stderr)
        return 1

    # Parse arguments
    staged = '--staged' in args or '--stage' in args or '-s' in args

    entries = read_index(repo_root)

    for entry in sorted(entries, key=lambda e: e['path']):
        if staged:
            # Format: mode sha stage path
            mode = entry['mode']
            sha = entry['sha']
            stage = 0  # Normal stage
            print(f"{mode} {sha} {stage}\t{entry['path']}")
        else:
            print(entry['path'])

    return 0

File: minigit-3/ts/src/commands/ls-files.ts
// ls-files command - List indexed files

import { findRepoRoot } from '../utils';
import { readIndex } from '../index-file';

export function lsFiles(args: string[]): void {
  const repoRoot = findRepoRoot();
  if (!repoRoot) {
    console.error('fatal: not a git repository');
    process.exit(1);
  }

  let showStaged = false;

  // Parse arguments
  for (const arg of args) {
    if (arg === '--stage' || arg === '--staged' || arg === '-s') {
      showStaged = true;
    }
  }

  const entries = readIndex(repoRoot);

  for (const entry of entries) {
    if (showStaged) {
      const mode = entry.mode.toString(8).padStart(6, '0');
      const stage = 0; // Stage 0 for normal entries
      console.log(`${mode} ${entry.sha} ${stage}\t${entry.name}`);
    } else {
      console.log(entry.name);
    }
  }
}
